### PR TITLE
LPD-33279 upgrade ddmField and ddmFieldAttribute company id with value

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
LPD-33279 JournalArticleDDMFieldsUpgradeProcess does not fill correctly the companyId Of the ddmFields and ddmFieldAttribute

# What is this trying to solve?

https://liferay.atlassian.net/browse/LPD-33279

`On the com.liferay.journal.internal.upgrade.v4_0_0.JournalArticleDDMFieldsUpgradeProcess` if needed, the company id on the creation of the `ddmField` and `ddmFieldAttribute` is obtained from the `CompanyThreadLocal.getCompanyId()` but because is a upgrade this value is 0.

# How am I fixing it?

On the `com.liferay.journal.internal.upgrade.v4_0_0.JournalArticleDDMFieldsUpgradeProcess` we should fill with the company data from the database.
And because the database can have data with values filled with 0 on `ddmField` and `ddmFieldAttribute`, we should create upgrades for each table with a process to fill this values with the values from the ddmStructure.

# How can you verify that it works?

For the new upgrades: Run the included automated tests.

# Why doesn't this include any test for the JournalArticleDDMFieldsUpgradeProcess?

I don't know how to reproduce it without the client data, **in addition** the failing upgrade is on a datatable version that have **2 columns** that master does not have.

# Commits

- **LPD-33279 populate the CompanyThreadLocal companyId value with the data from the query. We can not use the processConcurrently because the CompanyThreadLocal does not allow the modification on it.**
- **LPD-33279 create upgrade to fill the incorrect data that can be generated by error**
- **LPD-33279 register the new upgrades**
- **LPD-33279 add test to check the new upgrades**

### CODE REVIEW

**Comment by AliciaGarciaGarcia** [comment]
ci:test:upgrade-all

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_5_1/DDMFieldAttributeUpgradeProcess.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+				if (ddmStructureVersion == null) {
+					if (_log.isWarnEnabled()) {
+						_log.warn(
+							"DDMStructureVersion not found for field ID " +
+								resultSet.getLong("fieldId"));
+					}
+				}
+				else {
+					updatePreparedStatement.setLong(
```
Comment (jkappler) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_5_1/DDMFieldAttributeUpgradeProcess.java:68]: @AliciaGarciaGarcia, order should follow service.xml 😅. ctCollectionId -> fieldAttributeId -> companyId

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_5_1/DDMFieldUpgradeProcess.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+				if (ddmStructureVersion == null) {
+					if (_log.isWarnEnabled()) {
+						_log.warn(
+							"DDMStructureVersion not found for field ID " +
+								resultSet.getLong("fieldId"));
+					}
+				}
+				else {
+					updatePreparedStatement.setLong(
```
Comment (jkappler) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_5_1/DDMFieldUpgradeProcess.java:62]: @AliciaGarciaGarcia, same, ordering by service.xml 😅

**Review by jkappler — CHANGES_REQUESTED** [review]

**Comment by AliciaGarciaGarcia** [comment]
Hi @jkappler 

I will wait the upgrade-all to finish and make the suggested changes.

Thanks!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_5_1/DDMFieldAttributeUpgradeProcess.java (hunk 1 above)
Comment (jkappler) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_5_1/DDMFieldAttributeUpgradeProcess.java:68]: My bad @AliciaGarciaGarcia

**Review by jkappler — APPROVED** [review]

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:reevaluate:1317401_11195

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:test:upgrade

**Comment by AliciaGarciaGarcia** [comment]
rebasing with master

**Comment by AliciaGarciaGarcia** [comment]
ci:test:upgrade

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:report:60946799

**Comment by AliciaGarciaGarcia** [comment]
ci:report:60942033

**Comment by AliciaGarciaGarcia** [comment]
ci:test:stable

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:upgrade

**Comment by AliciaGarciaGarcia** [comment]
ci:report:61080005

**Comment by AliciaGarciaGarcia** [comment]
ci:report:61080005

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Pull request closed without merging**